microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -223,6 +242,133 @@ def verify_kdumpcrash_large_memory_auto_size(
+                f"Testcase only support Mariner-MSHV. Found: {node.os}"
+            )
+
+        mshv = node.tools[Ls].path_exists("/dev/mshv", sudo=True)
+        if not mshv:
+            raise SkippedException("MSHV not present")
+
+        log.info(f"chkbinpath: {chkbinpath}")
+        log.info(f"chkloaderpath: {chkloaderpath}")
+
+        # Copy and Extract CHK tar on node
+        grub_config_file = "/boot/grub2/grub.cfg"
+        chk_bin_dir = "chk_bin"
+        chk_bin_folder = f"/tmp/{chk_bin_dir}"
+        chk_bin_src_path = f"/tmp/{chk_bin_dir}.tar.gz"
+
+        chk_loader_dir = "chk_loader"
+        chk_loader_folder = f"/tmp/{chk_loader_dir}"
+        chk_loader_src_path = f"/tmp/{chk_loader_dir}.tar.gz"
+
+        # Copy artifacts on to the node
+        node.shell.copy(PurePath(chkbinpath), PurePath(chk_bin_src_path))
+        node.shell.copy(PurePath(chkloaderpath), PurePath(chk_loader_src_path))
+        node.execute("sudo ls -lrt /tmp/chk_loader")
+        node.execute("sudo ls -lrt /tmp/chk_bin")
+
+        tar = node.tools[Tar]
+        tar.extract(
+            file=chk_bin_src_path,
+            dest_dir=chk_bin_folder,
+            gzip=True,
+            sudo=True,
+        )
+        tar.extract(
+            file=chk_loader_src_path,
+            dest_dir=chk_loader_folder,
+            gzip=True,
+            sudo=True,
+        )
+
+        # Copy CHK bins into test machine
+        copy_tool = node.tools[Cp]
+        copy_tool.copy(
+            src=PurePath(chk_bin_folder) / "Windows" / "System32",
+            dest=PurePath("/boot/efi/Windows"),
+            sudo=True,
+            recur=True,
+        )
+        copy_tool.copy(
+            src=PurePath(chk_loader_folder) / "boot" / "efi" / "lxhvloader.dll",
+            dest=PurePath("/boot/efi"),
+            sudo=True,
+        )
+
+        # Remove kernel lockdown from grub config
+        node.tools[Sed].substitute(
+            regexp="lockdown=integrity",
+            replacement="",
+            file=grub_config_file,
+            sudo=True,
+        )
+
+        # Add MSHV debug option in chainloader
+        hv_debug_option = "LXHVLOADER_DEBUG=TRUE"
+        grub_config = node.tools[Cat].read(
+            file=grub_config_file,
+            force_run=True,
+            sudo=True,
+        )
+        regex = re.compile(r"(?P<chainloader_cfg>.*chainloader.*)")
+        chainloader_grub_data = find_group_in_lines(
+            lines=grub_config,
+            pattern=regex,
+            single_line=False,
+        )
+        chainloader_config = chainloader_grub_data.get("chainloader_cfg", "").strip()
+        err_msg = f"Cannot get chainloader config, got {chainloader_config}"
+        assert chainloader_config, err_msg
+        if hv_debug_option not in chainloader_config:
+            node.tools[Sed].substitute(
+                regexp="MSHV_SEV_SNP=TRUE",
+                replacement=f"MSHV_SEV_SNP=TRUE {hv_debug_option}",
```

Comment:
add comments for why need to add hv_debug_option

Reply:
Done. Added as comment before line `hv_debug_option = "LXHVLOADER_DEBUG=TRUE"`